Adding NinjaRSS module

pataluc opened the pull request:
Seemed useful to include the NinjaRSS module to this rep...
And I made a correction in the labels for "link to rss" option, "Yes" and "No" were inverted.

pataluc committed:
ajout module ninja rss

pataluc committed:
Yes and No labels where inverted regarding the link to feed option

GeraldST commented:
Hi pataluc,

Very nice to add the module - that really comes up handy. Anyway I fear there are some minor (warnings/notices) and also fatal errors in the code. Did you test it yourself? It also would be cool to have an installation package again (I just copied everything to my installation directory /modules/mod_ninja_rss_syndicator and then checked for new extensions in Backend).

Anyway I tested your code and came up with these errors that I could fix already, so I made comments on your code lines and would suggest to replace those lines. Here are the errors that I got:

Notice: Use of undefined constant DS - assumed 'DS' in /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on line 15

Fatal error: require_once(): Failed opening required '/joomla/modules/mod_ninja_rss_syndicatorDShelper.php' 

(include_path='.:/usr/lib/php5.4') in /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on line 15

Strict Standards: Non-static method modNinjaRssSyndicatorHelper::getNinjaRssSyndicatorFeeds() should not be called statically in 

/joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on line 22

Strict Standards: Only variables should be assigned by reference in /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 17

Warning: Creating default object from empty value in /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 44

Notice: Use of undefined constant DS - assumed 'DS' in /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 50

Strict Standards: Only variables should be assigned by reference in /joomla/modules/mod_ninja_rss_syndicator/tmpl/default.php on line 13

Warning: preg_match() expects parameter 2 to be string, array given in /joomla/modules/mod_ninja_rss_syndicator/tmpl/default.php on line 34

pataluc commented:
Hi Gerald,

i tested it very quickly, i'm not very surprised that it can be
perfected... :) i suggest to add you as collaborators on the repo, you
should be able to make the changes yourself (i have not the time currently
to do it myself...)

++
Luc

Le mar. 14 avr. 2015 à 23:59, GeraldST notifications@github.com a écrit :

> Hi pataluc,
> 
> Very nice to add the module - that really comes up handy. Anyway I fear
> there are some minor (warnings/notices) and also fatal errors in the code.
> Did you test it yourself? It also would be cool to have an installation
> package again (I just copied everything to my installation directory
> /modules/mod_ninja_rss_syndicator and then checked for new extensions in
> Backend).
> 
> Anyway I tested your code and came up with these errors that I could fix
> already, so I made comments on your code lines and would suggest to replace
> those lines. Here are the errors that I got:
> 
> Notice: Use of undefined constant DS - assumed 'DS' in
> /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on
> line 15
> 
> Fatal error: require_once(): Failed opening required
> '/joomla/modules/mod_ninja_rss_syndicatorDShelper.php'
> 
> (include_path='.:/usr/lib/php5.4') in
> /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on
> line 15
> 
> Strict Standards: Non-static method
> modNinjaRssSyndicatorHelper::getNinjaRssSyndicatorFeeds() should not be
> called statically in
> 
> /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on
> line 22
> 
> Strict Standards: Only variables should be assigned by reference in
> /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 17
> 
> Warning: Creating default object from empty value in
> /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 44
> 
> Notice: Use of undefined constant DS - assumed 'DS' in
> /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 50
> 
> Strict Standards: Only variables should be assigned by reference in
> /joomla/modules/mod_ninja_rss_syndicator/tmpl/default.php on line 13
> 
> Warning: preg_match() expects parameter 2 to be string, array given in
> /joomla/modules/mod_ninja_rss_syndicator/tmpl/default.php on line 34
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NinjaForge/Ninja-RSS-Syndicator/pull/9#issuecomment-93082411
> .

GeraldST commented:
Hi pataluc, thanks, I can do that. Actually tried it already but unfortunately Online Editor will show all lines as changed of the three files I need to touch (might have something to do with line endings). Is this okay? I mean no harm done with that only small issue with a lot of red/green lines that I would like to align with you first ;-)

pataluc commented:
no problem for me, you can do it...

Le mer. 15 avr. 2015 à 21:43, GeraldST notifications@github.com a écrit :

> Hi pataluc, thanks, I can do that. Actually tried it already but
> unfortunately Online Editor will show all lines as changed of the three
> files I need to touch (might have something to do with line endings). Is
> this okay? I mean no harm done with that only small issue with a lot of
> red/green lines that I would like to align with you first ;-)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NinjaForge/Ninja-RSS-Syndicator/pull/9#issuecomment-93544613
> .

GeraldST committed:
Fixed several errors

Strict Standards: Only variables should be assigned by reference in /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 17
Warning: Creating default object from empty value in /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 44
Notice: Use of undefined constant DS - assumed 'DS' in /joomla/modules/mod_ninja_rss_syndicator/helper.php on line 50

GeraldST committed:
Fixed Several errors

Notice: Use of undefined constant DS - assumed 'DS' in /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on line 15
Fatal error: require_once(): Failed opening required '/joomla/modules/mod_ninja_rss_syndicatorDShelper.php' 
(include_path='.:/usr/lib/php5.4') in /joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on line 15
Strict Standards: Non-static method modNinjaRssSyndicatorHelper::getNinjaRssSyndicatorFeeds() should not be called statically in 
/joomla/modules/mod_ninja_rss_syndicator/mod_ninja_rss_syndicator.php on line 22

GeraldST committed:
Fixed several errors

Strict Standards: Only variables should be assigned by reference in /joomla/modules/mod_ninja_rss_syndicator/tmpl/default.php on line 13
Warning: preg_match() expects parameter 2 to be string, array given in /joomla/modules/mod_ninja_rss_syndicator/tmpl/default.php on line 34

GeraldST committed:
Changed version

GeraldST commented:
Okay, all done!

GeraldST committed:
Solved wrong Syntax
